### PR TITLE
Add support for BucketKeyStatus to S3 inventories

### PR DESCRIPTION
**Pull request by orf**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: aws/resource_aws_s3_bucket_inventory.go [hunk 1]
```diff
@@ -177,6 +177,7 @@ func resourceAwsS3BucketInventory() *schema.Resource {
 						s3.InventoryOptionalFieldObjectLockRetainUntilDate,
 						s3.InventoryOptionalFieldObjectLockLegalHoldStatus,
 						s3.InventoryOptionalFieldIntelligentTieringAccessTier,
+						s3.InventoryOptionalFieldBucketKeyStatus,
```
Comment (ewbankkit) [inline aws/resource_aws_s3_bucket_inventory.go:180]: Please change to

```go
ValidateFunc: validation.StringInSlice(s3.InventoryOptionalField_Values(), false),
```

See #14601.

File: website/docs/r/s3_bucket_inventory.html.markdown [hunk 2]
```diff
@@ -89,7 +89,7 @@ The following arguments are supported:
 * `enabled` - (Optional, Default: `true`) Specifies whether the inventory is enabled or disabled.
 * `filter` - (Optional) Specifies an inventory filter. The inventory only includes objects that meet the filter's criteria (documented below).
 * `optional_fields` - (Optional) List of optional fields that are included in the inventory results.
-Valid values: `Size`, `LastModifiedDate`, `StorageClass`, `ETag`, `IsMultipartUploaded`, `ReplicationStatus`, `EncryptionStatus`, `ObjectLockRetainUntilDate`, `ObjectLockMode`, `ObjectLockLegalHoldStatus`, `IntelligentTieringAccessTier`.
+Valid values: `Size`, `LastModifiedDate`, `StorageClass`, `ETag`, `IsMultipartUploaded`, `ReplicationStatus`, `EncryptionStatus`, `ObjectLockRetainUntilDate`, `ObjectLockMode`, `ObjectLockLegalHoldStatus`, `IntelligentTieringAccessTier`, `BucketKeyStatus`.
```
Comment (ewbankkit) [inline website/docs/r/s3_bucket_inventory.html.markdown:92]: Please change to

```
* `optional_fields` - (Optional) List of optional fields that are included in the inventory results. Please refer to the S3 [documentation](https://docs.aws.amazon.com/AmazonS3/latest/API/API_InventoryConfiguration.html#AmazonS3-Type-InventoryConfiguration-OptionalFields) for more details.
```

File: aws/resource_aws_s3_bucket_inventory.go (hunk 1 above)
Comment (orf) [reply aws/resource_aws_s3_bucket_inventory.go]: Fixed! I was wondering about this

File: website/docs/r/s3_bucket_inventory.html.markdown (hunk 2 above)
Comment (orf) [reply website/docs/r/s3_bucket_inventory.html.markdown]: Fixed! Thanks

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccAWSS3BucketInventory_basic'              
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSS3BucketInventory_basic -timeout 180m
=== RUN   TestAccAWSS3BucketInventory_basic
=== PAUSE TestAccAWSS3BucketInventory_basic
=== CONT  TestAccAWSS3BucketInventory_basic
--- PASS: TestAccAWSS3BucketInventory_basic (141.21s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       148.518s
```

**Comment by ewbankkit** [comment]
@orf Thanks for the contribution :tada: :clap:.

**Pull request merged**